front/app/containers/Admin/projects/project/phase/phaseParticipationConfig/index.tsx

```diff
@@ -115,591 +65,490 @@ const MAX_VOTES_PER_VOTING_METHOD: Record<VotingMethod, number> = {
-          voting_method: newData.voting_method,
-          presentation_mode: newData.presentation_mode,
-          voting_min_total: newData.voting_min_total,
-          voting_max_total: newData.voting_max_total,
-          voting_max_votes_per_idea: newData.voting_max_votes_per_idea,
-          voting_term_plural_multiloc: newData.voting_term_plural_multiloc,
-          voting_term_singular_multiloc: newData.voting_term_singular_multiloc,
-          survey_embed_url: newData.survey_embed_url,
-          survey_service: newData.survey_service,
-          poll_anonymous: newData.poll_anonymous,
-          ideas_order: newData.ideas_order,
-          input_term: newData.input_term,
-          document_annotation_embed_url: newData.document_annotation_embed_url,
-          loaded: true,
-        };
-      });
-    } else {
-      this.setState({ loaded: true });
-    }
-
-    this.subscriptions = [
-      eventEmitter
-        .observeEvent('getPhaseParticipationConfig')
-        .pipe(filter(() => this.validate()))
-        .subscribe(() => {
-          const output = getOutput(this.state);
-          this.props.onSubmit(output);
-        }),
-    ];
-  }
-
-  componentDidUpdate(_prevProps: Props, prevState: State) {
-    const {
-      noLikingLimitError: _prevNoLikingLimit,
-      noDislikingLimitError: _prevNoDislikingLimit,
-      loaded: _prevLoaded,
-      ...prevPartialState
-    } = prevState;
-    const {
-      noLikingLimitError: _nextNoLikingLimit,
-      noDislikingLimitError: _nextNoDislikingLimit,
-      loaded: _nextLoaded,
-      ...nextPartialState
-    } = this.state;
-
-    if (!isEqual(prevPartialState, nextPartialState)) {
-      const output = getOutput(this.state);
-      this.props.onChange(output);
-    }
-
-    const { phase } = this.props;
-    if (
-      phase &&
-      _prevProps.phase?.data.attributes.participation_method !==
-        phase.data.attributes.participation_method
-    ) {
-      this.setState({
-        participation_method: phase.data.attributes.participation_method,
-      });
-    }
-  }
 
-  componentWillUnmount() {
-    this.subscriptions.forEach((subscription) => subscription.unsubscribe());
-  }
+    const subscription = eventEmitter
+      .observeEvent('getPhaseParticipationConfig')
+      .pipe(filter(() => validate()))
+      .subscribe(() => {
+        onSubmit(participationConfig);
+      });
+    return () => subscription.unsubscribe();
+  }, [participationConfig, onSubmit, formatMessage, appConfig]);
 
-  handleParticipationMethodOnChange = (
+  const handleParticipationMethodOnChange = (
     participation_method: ParticipationMethod
   ) => {
     const ideation = participation_method === 'ideation';
+    const native_survey = participation_method === 'native_survey';
     const voting = participation_method === 'voting';
     const survey = participation_method === 'survey';
-    const ideationOrVoting = ideation || voting;
 
-    this.setState({
+    setParticipationConfig(() => ({
```

Comment:
Most methods can rely on the default config but for the ones that need some specific attributes, we are transparently overriding it.